### PR TITLE
test(e2e): add digitale adressen fallback verification scenarios

### LINKED ISSUE #447: 🛠 Task: E2E tests voor fallback-logica digitale adressen partij
# 🛠 Task: E2E tests voor fallback-logica digitale adressen (Phase 2)

---

## Intent

Verifieer dat de fallback-logica voor digitale adressen via partij correct werkt in een gedeployde omgeving, zodat alle Gherkin-scenario's uit de implementatietaken aantoonbaar slagen als geautomatiseerde tests.

---

## Functional Description

Schrijf Playwright E2E tests die alle scenario's uit Feature #367 afdekken. De tests valideren dat:
- Partij-adressen worden getoond wanneer betrokkene geen eigen adressen heeft
- Eigen adressen van betrokkene altijd voorrang hebben
- Lege contactsectie wordt getoond wanneer geen adressen beschikbaar zijn
- Geen herkomst-indicatie zichtbaar is

Deze taak wordt opgepakt na deploy van de implementatietaken (#445, #446) naar de test-omgeving (`ita.test.icatt.nl`).

---

## Design Reference

N/A — geen design voor deze taak.

---

### In scope

- Playwright E2E tests voor alle Gherkin-scenario's uit Task #445 en #446
- Test data setup via `TestDataHelper` (betrokkene met/zonder partij, partij met/zonder adressen)
- Cleanup van testdata via `RegisterCleanup()`

### Out of scope

- Implementatiewijzigingen (reeds afgerond in #445 en #446)
- Unit tests / integration tests (niet in de teststrategie)
- Performance testing

---

### Request

N/A — tests gebruiken de bestaande UI en API flows.

### Response

N/A — tests valideren de visuele output in het behandelscherm.

### Events / Side Effects

Geen — tests zijn read-only verificatie.

---

## Technical Approach

- Maak een nieuwe test class `DigitaleAdressenFallbackScenarios` in `InterneTaakAfhandeling.EndToEndTest/Contactverzoek/`
- Inherit van `ITAPlaywrightTest`
- Gebruik `TestDataHelper` om testdata aan te maken:
  - Contactverzoek met betrokkene zonder eigen adressen + partij met adressen
  - Contactverzoek met betrokkene met eigen adressen + partij met andere adressen
  - Contactverzoek met betrokkene zonder adressen en zonder partij
  - Contactverzoek met betrokkene zonder adressen + partij zonder adressen
- Navigeer naar behandelscherm en verifieer zichtbaarheid van contactgegevens
- Gebruik `RegisterCleanup()` voor testdata cleanup

### Aggregates & Invariants

- Aggregates: Geen (test-only)
- Invariants: Tests moeten idempotent en onafhankelijk van elkaar uitvoerbaar zijn

---

## Impacted Areas

- Backend: Geen
- Frontend: Geen
- Database: N/A
- APIs: Geen wijzigingen — tests consumeren bestaande endpoints
- External systems: Test-omgeving `ita.test.icatt.nl`
- Tests: `InterneTaakAfhandeling.EndToEndTest/Contactverzoek/DigitaleAdressenFallbackScenarios.cs`

---

## Gherkin

Referentie naar scenario's uit #445 en #446:

```gherkin
Feature: E2E verificatie fallback digitale adressen

  Scenario: Partij-adressen getoond bij afwezigheid eigen adressen betrokkene
    Given een contactverzoek waarbij de betrokkene geen eigen digitale adressen heeft
    And de betrokkene is gekoppeld aan een partij met een e-mailadres en telefoonnummer
    When de medewerker het contactverzoek opent in het behandelscherm
    Then zijn het e-mailadres en telefoonnummer zichtbaar in de contactgegevens

  Scenario: Eigen adressen betrokkene hebben voorrang
    Given een contactverzoek waarbij de betrokkene een eigen e-mailadres heeft
    And de betrokkene is gekoppeld aan een partij met een ander e-mailadres
    When de medewerker het contactverzoek opent in het behandelscherm
    Then is alleen het eigen e-mailadres van de betrokkene zichtbaar

  Scenario: Lege contactsectie zonder adressen
    Given een contactverzoek waarbij de betrokkene geen eigen digitale adressen heeft
    And de betrokkene is niet gekoppeld aan een partij
    When de medewerker het contactverzoek opent in het behandelscherm
    Then is de contactsectie leeg

  Scenario: Lege contactsectie met partij zonder adressen
    Given een contactverzoek waarbij de betrokkene geen eigen digitale adressen heeft
    And de betrokkene is gekoppeld aan een partij zonder digitale adressen
    When de medewerker het contactverzoek opent in het behandelscherm
    Then is de contactsectie leeg

  Scenario: Geen herkomst-indicatie zichtbaar
    Given een contactverzoek waarbij partij-adressen worden getoond via fallback
    When de medewerker het contactverzoek opent in het behandelscherm
    Then is er geen indicatie zichtbaar dat de adressen van de partij afkomstig zijn
```

---

## Edge Cases & Risks

- Test-omgeving moet beschikbaar zijn met de gedeployde fallback-functionaliteit
- Testdata cleanup moet betrouwbaar werken om test-isolatie te garanderen
- Session timeouts bij langlopende testreeks (bekende flaky area per QA.md)

---

## Observability

- Logs: Playwright test output met `Step()` annotaties
- Metrics: Geen
- Alerts: Geen

---

## Dependencies

- Depends on #445: API client methode moet geïmplementeerd en gedeployd zijn
- Depends on #446: Fallback-logica moet geïmplementeerd en gedeployd zijn
- Requires deployment to test environment (`ita.test.icatt.nl`)

---

## Rollout

- Feature flag: N/A
- Backward compatibility: N/A (test-only)
- Data migration: N/A

---

## Definition of Done

- [x] All Gherkin scenarios covered by automated E2E tests
- [x] E2E tests pass against the test environment
- [x] No new flaky tests introduced
- [x] Edge cases listed in the Feature are covered by E2E tests
- [ ] Code reviewed
- [ ] PR approved by at least one reviewer

---

## Test Mapping

| Gherkin Scenario | Test file | Status |
| ---------------- | --------- | ------ |
| Partij-adressen getoond bij afwezigheid eigen adressen | DigitaleAdressenFallbackScenarios.cs | Pending |
| Eigen adressen betrokkene hebben voorrang | DigitaleAdressenFallbackScenarios.cs | Pending |
| Lege contactsectie zonder adressen | DigitaleAdressenFallbackScenarios.cs | Pending |
| Lege contactsectie met partij zonder adressen | DigitaleAdressenFallbackScenarios.cs | Pending |
| Geen herkomst-indicatie zichtbaar | DigitaleAdressenFallbackScenarios.cs | Pending |

### PR DESCRIPTION
**Pull request by nijmra**
## Summary

Verifies that the partij digitale adressen fallback logic (implemented in #445 and #446) works correctly in the deployed test environment. Ensures that when a betrokkene has no own digital addresses, the system falls back to the linked partij's addresses — and that own addresses always take priority.

Part of Feature #367.

## Changes

- **New test class `DigitaleAdressenFallbackScenarios`**: 4 E2E scenarios covering partij fallback, own-address priority, empty state (no partij), and empty state (partij without addresses)
- **TestDataHelper extensions**: new methods for creating contactverzoeken with specific partij configurations, managing digitale adressen on betrokkenen and partijen, and creating/deleting partijen without addresses
- **Infrastructure alignment**: adapted to use unique internetaakNummer pattern and removed NetworkIdle waits (consistent with fix/e2e-test-flakiness-2 base)

## Test plan

- [x] Partij-adressen getoond bij afwezigheid eigen adressen betrokkene
- [x] Eigen adressen betrokkene hebben voorrang
- [x] Lege contactsectie zonder adressen (betrokkene niet gekoppeld aan partij)
- [x] Lege contactsectie met partij zonder adressen
- [x] Geen herkomst-indicatie zichtbaar

## Related

Closes #447

### CODE REVIEW

File: InterneTaakAfhandeling.EndToEndTest/Infrastructure/TestDataHelper.cs [hunk 1]
```diff
@@ -738,5 +748,223 @@ private Task CreateOnderwerpobjectForZaak(Guid klantcontactUuid, string zaakUuid
...
+                voorkeursDigitaalAdres = (object?)null,
+                rpipiereesvertegenwoordigerVan = Array.Empty<object>(),
+                partijIdentificatoren = Array.Empty<object>(),
+                soortPartij = "persoon",
+                indicatieActief = true,
+                indicatieGeheimhouding = false,
+                voorkeurstaal = "nl",
+                bezoekadres = (object?)null,
+                correspondentieadres = (object?)null
+            };
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.EndToEndTest/Infrastructure/TestDataHelper.cs:904]: Fixed — removed the misspelled field entirely. It's not required for a minimal partij creation.

File: InterneTaakAfhandeling.EndToEndTest/Infrastructure/TestDataHelper.cs [hunk 2]
```diff
@@ -738,5 +748,223 @@ private Task CreateOnderwerpobjectForZaak(Guid klantcontactUuid, string zaakUuid
...
+                partij.Naam);
+
+            var betrokkene = await AttachPartijToContactmoment(Guid.Parse(partij.Uuid), contactmoment.Uuid);
+
+            // Create own digitale adres on the betrokkene
+            await CreateDigitaalAdresForBetrokkeneAsync(
+                betrokkene.Uuid,
+                ownEmail,
+                "email",
+                "Eigen e-mail");
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.EndToEndTest/Infrastructure/TestDataHelper.cs]: Fixed — `CreateContactverzoekWithPartijAndOwnAdressenAsync` now returns the `digitaalAdresUuid` and the test registers a cleanup for it.

File: InterneTaakAfhandeling.EndToEndTest/Contactverzoek/DigitaleAdressenFallbackScenarios.cs [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+        [TestMethod("Eigen adressen betrokkene hebben voorrang boven partij-adressen")]
+        public async Task Medewerker_ZietEigenAdressen_WanneerBetrokkeneEigenAdressenHeeft()
+        {
+            var onderwerp = $"E2E_Fallback_EigenAdressen_{DateTime.UtcNow.Ticks}";
+            var ownEmail = "eigen-test@e2e-ita.nl";
+
+            await Step("Setup contactverzoek with partij AND own digitale adressen on betrokkene");
+            var (contactmomentUuid, _) = await TestDataHelper.CreateContactverzoekWithPartijAndOwnAdressenAsync(
+                onderwerp, ownEmail);
+            RegisterCleanup(async () => await TestDataHelper.DeleteContactverzoekAsync(contactmomentUuid.ToString()));
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.EndToEndTest/Contactverzoek/DigitaleAdressenFallbackScenarios.cs:130]: Fixed — the test now ensures the partij has an email before testing priority logic (same pattern as `SetupAndNavigateToContactverzoekWithPartij`).

File: InterneTaakAfhandeling.EndToEndTest/Contactverzoek/DigitaleAdressenFallbackScenarios.cs [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+        [TestMethod("Lege contactsectie wanneer partij geen adressen heeft")]
+        public async Task Medewerker_ZietLegeContactsectie_WanneerPartijGeenAdressenHeeft()
+        {
+            // This scenario requires a partij without digitale adressen in the test environment.
+            // The unit tests (InternetaakDetailsServiceFallbackTests.Get_ReturnsEmptyAdressen_WhenPartijHasNoAdressen)
+            // cover the server-side behavior. This E2E test validates the full stack.
+            //
+            // Setup: uses a partij that exists in OpenKlant but has no digitale adressen.
+            // If no such partij is available, this test should be marked Inconclusive.
+
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.EndToEndTest/Contactverzoek/DigitaleAdressenFallbackScenarios.cs:184]: Fixed — removed the references to non-existent unit tests from the comments.

File: InterneTaakAfhandeling.EndToEndTest/Contactverzoek/DigitaleAdressenFallbackScenarios.cs [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+            try
+            {
+                partijUuid = await TestDataHelper.CreatePartijWithoutAdressenAsync();
+                RegisterCleanup(async () => await TestDataHelper.DeletePartijAsync(partijUuid));
+            }
+            catch (Exception ex)
+            {
+                Assert.Inconclusive(
+                    $"Could not create partij without adressen in test environment: {ex.Message}. " +
+                    "This scenario is covered by unit tests.");
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.EndToEndTest/Contactverzoek/DigitaleAdressenFallbackScenarios.cs]: Fixed — removed the misleading unit test reference from the Inconclusive message.

File: InterneTaakAfhandeling.EndToEndTest/Contactverzoek/DigitaleAdressenFallbackScenarios.cs [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+                await assertions();
+            }
+            catch (PlaywrightException)
+            {
+                await Step("Retrying after page reload (external API may be slow)");
+                await Page.ReloadAsync(new() { WaitUntil = WaitUntilState.DOMContentLoaded });
+                await Expect(Page.GetContactmomentRegistrerenTab()).ToBeVisibleAsync();
+                await assertions();
+            }
+        }
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.EndToEndTest/Contactverzoek/DigitaleAdressenFallbackScenarios.cs:77]: Won't fix — the Playwright `Expect().ToHaveTextAsync()` calls already handle the transient wait via their built-in timeout. The raw `Assert.*` calls are for immediate structural checks where a retry wouldn't change the outcome.

**Review by surbhijha92 — APPROVED** [review]
